pkg/machines/components/storagePools/storagePoolDelete.jsx

```diff
@@ -86,33 +131,83 @@ export class StoragePoolDelete extends React.Component {
     }
 
     render() {
-        const { storagePool } = this.props;
+        const { storagePool, vms } = this.props;
         const id = storagePoolId(storagePool.name, storagePool.connectionName);
         const volumes = storagePool.volumes || [];
 
+        const usage = getStorageVolumesUsage(vms, storagePool);
+        let vmsUsage = [];
+        for (let property in usage)
+            vmsUsage = vmsUsage.concat(usage[property]);
+
+        vmsUsage = vmsUsage.join(', ');
+        const showWarning = () => {
+            if (canDeleteOnlyWithoutVolumes(storagePool, vms) && this.state.deleteVolumes) {
+                return (
+                    <span id={`delete-${id}-idle-message`}>
+                        <i className='pficon pficon-info' />
+                        {_("Pool's volumes are used by: ") + vmsUsage + '. '}
+                        {_("Detach the disks which use this pool from any VMs before attempting deletion.")}
+                    </span>
+                );
+            }
+        };
+
         let defaultBody = (
-            <div className='ct-form'>
-                { storagePool.active && volumes.length > 0 && <React.Fragment>
-                    <label className='control-label'>
-                        {_("Delete Content")}
-                    </label>
-                    <label className='checkbox-inline'>
-                        <input id='storage-pool-delete-volumes'
-                            type='checkbox'
-                            checked={this.state.deleteVolumes}
-                            onChange={e => this.onValueChanged('deleteVolumes', e.target.checked)} />
-                        {_("Delete the Volumes inside this Pool")}
-                    </label>
-                </React.Fragment>}
-                { !storagePool.active && _("Deleting an inactive Storage Pool will only undefine the Pool. Its content will not be deleted.")}
-            </div>
+            <React.Fragment>
+                <div className='ct-form'>
+                    { storagePool.active && volumes.length > 0 && <React.Fragment>
+                        <label className='control-label'>
+                            {_("Delete Content")}
```

Comment:
I don't think we need this label, since it doesn't add much information.
I would just show the checkbox and the label.

Reply:
This label was already there (I just changed code alignment), so I would say that should go into another PR.